bindgen/private/bindgen.bzl

```diff
@@ -129,26 +129,46 @@ def _rust_bindgen_impl(ctx):
 
     # Configure Clang Arguments
     args.add("--")
-    args.add_all(include_directories, before_each = "-I")
-    args.add_all(quote_include_directories, before_each = "-iquote")
-    args.add_all(system_include_directories, before_each = "-isystem")
-    args.add_all(ctx.attr.clang_flags)
 
-    cc_toolchain, feature_configuration = find_cc_toolchain(ctx)
+    cc_toolchain, feature_configuration = find_cc_toolchain(ctx = ctx)
+    compile_variables = cc_common.create_compile_variables(
+        cc_toolchain = cc_toolchain,
+        feature_configuration = feature_configuration,
+        include_directories = cc_lib[CcInfo].compilation_context.includes,
+        quote_include_directories = cc_lib[CcInfo].compilation_context.quote_includes,
+        system_include_directories = cc_lib[CcInfo].compilation_context.system_includes,
+        user_compile_flags = ctx.attr.clang_flags,
+    )
+    compile_flags = cc_common.get_memory_inefficient_command_line(
+        feature_configuration = feature_configuration,
+        action_name = CPP_COMPILE_ACTION_NAME,
+        variables = compile_variables,
+    )
+
+    # Parse known arguments from the list of potential compile flags
+    open_arg = False
+    for arg in compile_flags:
```

Comment:
Why are we filtering these, rather than just appending all of them?

Reply:
My hope was to allow the toolchain to specify the order of the flags

Reply:
What's stopping us from just making this whole `for`-loop instead just be:

```
args.add_all(compile_flags)
args.add_all(ctx.attr.clang_flags)
```

I don't think I'm understanding the feature stuff enough to grasp this - are there specific features' flags we're trying to include/exclude which this for-loop-plus-filter is a reasonable proxy for?

Reply:
I originally tried that and then got bullied by non clang flags. I would love to do this eventually but am nervous to roll out too large a change at once since I don't feel like these rules are that well understood.

Reply:
Aha, I see - sounds good; I just pushed a few comments to your branch, PTAL and merge if you're happy!

Reply:
Love it! Thanks!